### PR TITLE
Add share links in blog posts pages

### PR DESCRIPTION
**Pull request by pfac**
Why:

* As designed.

This change addresses the need by:

* Adding `react-share`, which seems to be the standard dependency to add
  social sharing links;
* Pre-computing the URL for each blog post after the Markdown node is
  created, relying on the `URL` environment variable which is set on
  Netlify to use the default site name or the custom domain the site was
  set up with (more info in the [Netlify docs]);
* Refactoring the wrapper components used in a blog post to accept a
  custom `className` from the parent components;
* Refactoring the blog post layout to place the sharing links on the
  left of the text, as designed, falling back to the vertical flow at
  the end of the post in mobile.

[Netlify docs]: https://docs.netlify.com/configure-builds/environment-variables/#git-metadata

### CODE REVIEW

File: src/components/blog/post/body/share_links.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React from "react"
+import PropTypes from "prop-types"
+import classNames from "classnames"
+import {
+  FacebookShareButton,
+  LinkedinShareButton,
+  TwitterShareButton,
+} from "react-share"
+
+import useDetectJavascript from "../../../../utils/use_detect_javascript"
```
Comment (jfranciscosousa) [inline src/components/blog/post/body/share_links.js:10]: This pains me. One day try and refactor into using a `root` alias, not required for now

File: src/components/blog/post/body/share_links.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import styles from "./share_links.module.scss"
+
+const BlogPostShareLinks = ({ className, url }) => {
+  const hasJavascript = useDetectJavascript()
```
Comment (jfranciscosousa) [inline src/components/blog/post/body/share_links.js:15]: Also, I was a bit stupid when I named this. A good alternative should be `hasMounted`, it's the step after the client rehydrates the server-rendered React page, so just keep that in mind

File: src/components/blog/post/body/share_links.js (hunk 1 above)
Comment (pfac) [reply src/components/blog/post/body/share_links.js:10]: What do you mean with `root` alias?

Comment (jfranciscosousa) [reply src/components/blog/post/body/share_links.js:10]: You can use aliases on `yarn/webpack/gatsbyplugin` to reference the root of the project like `root/utils/use_detect_javascript`

**Pull request merged**